flink-formats/flink-avro-glue-schema-registry/pom.xml

```diff
@@ -0,0 +1,99 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<!--
+Licensed to the Apache Software Foundation (ASF) under one
+or more contributor license agreements.  See the NOTICE file
+distributed with this work for additional information
+regarding copyright ownership.  The ASF licenses this file
+to you under the Apache License, Version 2.0 (the
+"License"); you may not use this file except in compliance
+with the License.  You may obtain a copy of the License at
+
+  http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing,
+software distributed under the License is distributed on an
+"AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+KIND, either express or implied.  See the License for the
+specific language governing permissions and limitations
+under the License.
+-->
+<project xmlns="http://maven.apache.org/POM/4.0.0"
+		 xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+		 xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
+	<parent>
+		<artifactId>flink-formats</artifactId>
+		<groupId>org.apache.flink</groupId>
+		<version>1.13-SNAPSHOT</version>
+		<relativePath>..</relativePath>
+	</parent>
+	<modelVersion>4.0.0</modelVersion>
+
+	<artifactId>flink-avro-glue-schema-registry</artifactId>
+	<name>Flink : Formats : Avro AWS Glue Schema Registry</name>
+	<packaging>jar</packaging>
+
+	<properties>
+		<glue.schema.registry.version>1.0.0</glue.schema.registry.version>
+		<junit.jupiter.version>5.6.2</junit.jupiter.version>
+		<enforcer.skip>true</enforcer.skip>
+	</properties>
+
+	<dependencies>
```

Comment:
I have taken a look at the dependency footprint of this module and it looks like there is too much pulled in:

Why do we need Kafka dependencies?
- `+- org.apache.kafka:connect-json:jar:2.5.0:compile`
- `+- org.apache.kafka:connect-api:jar:2.5.0:compile`
- `+- org.apache.kafka:kafka-streams:jar:2.5.0:compile`
- `+- org.apache.kafka:kafka-clients:jar:2.5.0:compile`

Pulling lombok as a compile dependency looks wrong, is this scoped correctly in upstream module?
- `+- org.projectlombok:lombok:jar:1.18.2:compile`
- `\- org.projectlombok:lombok-utils:jar:1.18.12:compile`

As mentioned, we should use the standard junit framework for flink:
- `+- org.junit.jupiter:junit-jupiter-api:jar:5.6.2:test`

```
[INFO] --- maven-dependency-plugin:3.1.1:tree (default-cli) @ flink-avro-glue-schema-registry ---
[INFO] org.apache.flink:flink-avro-glue-schema-registry:jar:1.13-SNAPSHOT
[INFO] +- org.apache.flink:flink-core:jar:1.13-SNAPSHOT:provided
[INFO] |  +- org.apache.flink:flink-annotations:jar:1.13-SNAPSHOT:provided
[INFO] |  +- org.apache.flink:flink-metrics-core:jar:1.13-SNAPSHOT:provided
[INFO] |  +- org.apache.flink:flink-shaded-asm-7:jar:7.1-12.0:provided
[INFO] |  +- org.apache.commons:commons-lang3:jar:3.3.2:compile
[INFO] |  +- com.esotericsoftware.kryo:kryo:jar:2.24.0:provided
[INFO] |  |  \- com.esotericsoftware.minlog:minlog:jar:1.2:provided
[INFO] |  +- commons-collections:commons-collections:jar:3.2.2:provided
[INFO] |  +- org.apache.commons:commons-compress:jar:1.20:compile
[INFO] |  \- org.apache.flink:flink-shaded-guava:jar:18.0-12.0:compile
[INFO] +- org.apache.flink:flink-avro:jar:1.13-SNAPSHOT:compile
[INFO] |  \- org.apache.avro:avro:jar:1.10.0:compile
[INFO] |     +- com.fasterxml.jackson.core:jackson-core:jar:2.12.1:compile
[INFO] |     \- com.fasterxml.jackson.core:jackson-databind:jar:2.12.1:compile
[INFO] |        \- com.fasterxml.jackson.core:jackson-annotations:jar:2.12.1:compile
[INFO] +- org.apache.flink:flink-streaming-java_2.11:jar:1.13-SNAPSHOT:compile
[INFO] |  +- org.apache.flink:flink-file-sink-common:jar:1.13-SNAPSHOT:compile
[INFO] |  +- org.apache.flink:flink-runtime_2.11:jar:1.13-SNAPSHOT:compile
[INFO] |  |  +- org.apache.flink:flink-queryable-state-client-java:jar:1.13-SNAPSHOT:compile
[INFO] |  |  +- org.apache.flink:flink-hadoop-fs:jar:1.13-SNAPSHOT:compile
[INFO] |  |  +- commons-io:commons-io:jar:2.7:compile
[INFO] |  |  +- org.apache.flink:flink-shaded-netty:jar:4.1.49.Final-12.0:compile
[INFO] |  |  +- org.apache.flink:flink-shaded-jackson:jar:2.10.1-12.0:compile
[INFO] |  |  +- org.apache.flink:flink-shaded-zookeeper-3:jar:3.4.14-12.0:compile
[INFO] |  |  +- org.javassist:javassist:jar:3.24.0-GA:compile
[INFO] |  |  +- org.scala-lang:scala-library:jar:2.11.12:compile
[INFO] |  |  +- com.typesafe.akka:akka-actor_2.11:jar:2.5.21:compile
[INFO] |  |  |  +- com.typesafe:config:jar:1.3.0:compile
[INFO] |  |  |  \- org.scala-lang.modules:scala-java8-compat_2.11:jar:0.7.0:compile
[INFO] |  |  +- com.typesafe.akka:akka-stream_2.11:jar:2.5.21:compile
[INFO] |  |  |  +- org.reactivestreams:reactive-streams:jar:1.0.2:compile
[INFO] |  |  |  \- com.typesafe:ssl-config-core_2.11:jar:0.3.7:compile
[INFO] |  |  |     \- org.scala-lang.modules:scala-parser-combinators_2.11:jar:1.1.1:compile
[INFO] |  |  +- com.typesafe.akka:akka-protobuf_2.11:jar:2.5.21:compile
[INFO] |  |  +- com.typesafe.akka:akka-slf4j_2.11:jar:2.5.21:compile
[INFO] |  |  +- org.clapper:grizzled-slf4j_2.11:jar:1.3.2:compile
[INFO] |  |  +- com.github.scopt:scopt_2.11:jar:3.5.0:compile
[INFO] |  |  +- org.xerial.snappy:snappy-java:jar:1.1.4:compile
[INFO] |  |  +- com.twitter:chill_2.11:jar:0.7.6:compile
[INFO] |  |  |  \- com.twitter:chill-java:jar:0.7.6:compile
[INFO] |  |  \- org.lz4:lz4-java:jar:1.6.0:compile
[INFO] |  +- org.apache.flink:flink-java:jar:1.13-SNAPSHOT:compile
[INFO] |  \- org.apache.commons:commons-math3:jar:3.5:compile
[INFO] +- org.apache.flink:flink-clients_2.11:jar:1.13-SNAPSHOT:compile
[INFO] |  +- org.apache.flink:flink-optimizer_2.11:jar:1.13-SNAPSHOT:compile
[INFO] |  \- commons-cli:commons-cli:jar:1.3.1:compile
[INFO] +- software.amazon.glue:schema-registry-serde:jar:1.0.0:compile
[INFO] |  +- software.amazon.glue:schema-registry-common:jar:1.0.0:compile
[INFO] |  |  +- software.amazon.awssdk:glue:jar:2.15.32:compile
[INFO] |  |  |  +- software.amazon.awssdk:protocol-core:jar:2.15.32:compile
[INFO] |  |  |  +- software.amazon.awssdk:auth:jar:2.15.32:compile
[INFO] |  |  |  |  \- software.amazon.eventstream:eventstream:jar:1.0.1:compile
[INFO] |  |  |  +- software.amazon.awssdk:http-client-spi:jar:2.15.32:compile
[INFO] |  |  |  +- software.amazon.awssdk:regions:jar:2.15.32:compile
[INFO] |  |  |  +- software.amazon.awssdk:aws-core:jar:2.15.32:compile
[INFO] |  |  |  +- software.amazon.awssdk:metrics-spi:jar:2.15.32:compile
[INFO] |  |  |  +- software.amazon.awssdk:apache-client:jar:2.15.32:runtime
[INFO] |  |  |  |  +- org.apache.httpcomponents:httpclient:jar:4.5.3:runtime
[INFO] |  |  |  |  |  +- commons-logging:commons-logging:jar:1.1.3:runtime
[INFO] |  |  |  |  |  \- commons-codec:commons-codec:jar:1.13:runtime
[INFO] |  |  |  |  \- org.apache.httpcomponents:httpcore:jar:4.4.6:runtime
[INFO] |  |  |  \- software.amazon.awssdk:netty-nio-client:jar:2.15.32:runtime
[INFO] |  |  |     +- io.netty:netty-codec-http:jar:4.1.53.Final:runtime
[INFO] |  |  |     +- io.netty:netty-codec-http2:jar:4.1.53.Final:runtime
[INFO] |  |  |     +- io.netty:netty-codec:jar:4.1.53.Final:runtime
[INFO] |  |  |     +- io.netty:netty-transport:jar:4.1.53.Final:runtime
[INFO] |  |  |     |  \- io.netty:netty-resolver:jar:4.1.53.Final:runtime
[INFO] |  |  |     +- io.netty:netty-common:jar:4.1.53.Final:runtime
[INFO] |  |  |     +- io.netty:netty-buffer:jar:4.1.53.Final:runtime
[INFO] |  |  |     +- io.netty:netty-handler:jar:4.1.53.Final:runtime
[INFO] |  |  |     +- io.netty:netty-transport-native-epoll:jar:linux-x86_64:4.1.53.Final:runtime
[INFO] |  |  |     |  \- io.netty:netty-transport-native-unix-common:jar:4.1.53.Final:runtime
[INFO] |  |  |     \- com.typesafe.netty:netty-reactive-streams-http:jar:2.0.4:runtime
[INFO] |  |  |        \- com.typesafe.netty:netty-reactive-streams:jar:2.0.4:runtime
[INFO] |  |  +- software.amazon.awssdk:aws-json-protocol:jar:2.15.30:compile
[INFO] |  |  +- software.amazon.awssdk:cloudwatch:jar:2.15.30:compile
[INFO] |  |  |  \- software.amazon.awssdk:aws-query-protocol:jar:2.15.30:compile
[INFO] |  |  +- software.amazon.awssdk:sdk-core:jar:2.15.30:compile
[INFO] |  |  |  \- software.amazon.awssdk:profiles:jar:2.15.30:compile
[INFO] |  |  +- org.apache.kafka:kafka-clients:jar:2.5.0:compile
[INFO] |  |  |  \- com.github.luben:zstd-jni:jar:1.4.4-7:compile
[INFO] |  |  +- org.apache.kafka:kafka-streams:jar:2.5.0:compile
[INFO] |  |  |  +- org.apache.kafka:connect-json:jar:2.5.0:compile
[INFO] |  |  |  |  +- org.apache.kafka:connect-api:jar:2.5.0:compile
[INFO] |  |  |  |  \- com.fasterxml.jackson.datatype:jackson-datatype-jdk8:jar:2.12.1:compile
[INFO] |  |  |  \- org.rocksdb:rocksdbjni:jar:5.18.3:compile
[INFO] |  |  \- com.google.guava:guava:jar:29.0-jre:compile
[INFO] |  |     +- com.google.guava:failureaccess:jar:1.0.1:compile
[INFO] |  |     +- com.google.guava:listenablefuture:jar:9999.0-empty-to-avoid-conflict-with-guava:compile
[INFO] |  |     +- org.checkerframework:checker-qual:jar:2.11.1:compile
[INFO] |  |     +- com.google.errorprone:error_prone_annotations:jar:2.3.4:compile
[INFO] |  |     \- com.google.j2objc:j2objc-annotations:jar:1.3:compile
[INFO] |  +- software.amazon.awssdk:arns:jar:2.15.26:compile
[INFO] |  |  +- software.amazon.awssdk:annotations:jar:2.15.26:compile
[INFO] |  |  \- software.amazon.awssdk:utils:jar:2.15.26:compile
[INFO] |  +- org.projectlombok:lombok:jar:1.18.2:compile
[INFO] |  \- org.projectlombok:lombok-utils:jar:1.18.12:compile
[INFO] +- org.junit.jupiter:junit-jupiter-api:jar:5.6.2:test
[INFO] |  +- org.apiguardian:apiguardian-api:jar:1.1.0:test
[INFO] |  +- org.opentest4j:opentest4j:jar:1.2.0:test
[INFO] |  \- org.junit.platform:junit-platform-commons:jar:1.6.2:test
[INFO] +- org.junit.jupiter:junit-jupiter-params:jar:5.6.2:test
[INFO] +- org.mockito:mockito-junit-jupiter:jar:2.21.0:test
[INFO] +- org.slf4j:slf4j-api:jar:1.7.15:provided
[INFO] +- org.apache.flink:flink-test-utils-junit:jar:1.13-SNAPSHOT:test
[INFO] +- org.apache.flink:force-shading:jar:1.13-SNAPSHOT:compile
[INFO] +- com.google.code.findbugs:jsr305:jar:1.3.9:compile
[INFO] +- junit:junit:jar:4.12:test
[INFO] |  \- org.hamcrest:hamcrest-core:jar:1.3:test
[INFO] +- org.mockito:mockito-core:jar:2.21.0:test
[INFO] |  +- net.bytebuddy:byte-buddy:jar:1.8.15:test
[INFO] |  +- net.bytebuddy:byte-buddy-agent:jar:1.8.15:test
[INFO] |  \- org.objenesis:objenesis:jar:2.1:provided
[INFO] +- org.powermock:powermock-module-junit4:jar:2.0.4:test
[INFO] |  \- org.powermock:powermock-module-junit4-common:jar:2.0.4:test
[INFO] |     +- org.powermock:powermock-reflect:jar:2.0.4:test
[INFO] |     \- org.powermock:powermock-core:jar:2.0.4:test
[INFO] +- org.powermock:powermock-api-mockito2:jar:2.0.4:test
[INFO] |  \- org.powermock:powermock-api-support:jar:2.0.4:test
[INFO] +- org.hamcrest:hamcrest-all:jar:1.3:test
[INFO] +- org.apache.logging.log4j:log4j-slf4j-impl:jar:2.12.1:test
[INFO] +- org.apache.logging.log4j:log4j-api:jar:2.12.1:test
[INFO] +- org.apache.logging.log4j:log4j-core:jar:2.12.1:test
[INFO] \- org.apache.logging.log4j:log4j-1.2-api:jar:2.12.1:test
```

Reply:
The enforcer check skip property has been removed. Currently, version of all dependencies with convergence error have been specifically defined in `Flink-GSR` module and its e2e test module. Once the new version of GSR package with reorganized dependencies is released to maven, the version definition can be removed.

Reply:
This comment looks misplaced. This is not related to the enforcer skip, it is looking at the transitive dependency chain. Did you reply to the wrong thread?

Reply:
The dependency chain is fixed in GSR package but it'll need some time to release. Once it's out, it should also fixed the enforcer check issue.

Reply:
ok, so we have removed the dependency on Kafka in the new version? What is the ECD for the new version?

Reply:
We will fix this in a follow up